Can't install any python version from continuum (miniconda | anaconda)

I am running:
macOS 10.12
pyenv (pyenv 1.0.6) and pyenv-virtualenv (pyenv-virtualenv 1.0.0 (virtualenv unknown)) were installed using homebrew

Installing a new version of python using `pyenv install ...` works fine. Installing miniconda or anaconda doesn't work. The error message is below.

```
bash-3.2$ pyenv install -v miniconda3-4.1.11
/var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229185759.9119 ~
Downloading Miniconda3-4.1.11-MacOSX-x86_64.sh...
-> https://repo.continuum.io/miniconda/Miniconda3-4.1.11-MacOSX-x86_64.sh
/var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229185759.9119/Miniconda3-4.1.11-MacOSX-x86_64 /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229185759.9119 ~
Installing Miniconda3-4.1.11-MacOSX-x86_64...
PREFIX=/Users/sly/.pyenv/versions/miniconda3-4.1.11
installing: python-3.5.2-0 ...
installing: conda-env-2.5.2-py35_0 ...
installing: openssl-1.0.2h-1 ...
installing: pycosat-0.6.1-py35_1 ...
installing: pyyaml-3.11-py35_4 ...
installing: readline-6.2-2 ...
installing: requests-2.10.0-py35_0 ...
installing: ruamel_yaml-0.11.14-py35_0 ...
installing: sqlite-3.13.0-0 ...
installing: tk-8.5.18-0 ...
installing: xz-5.2.2-0 ...
installing: yaml-0.1.6-0 ...
installing: zlib-1.2.8-3 ...
installing: conda-4.1.11-py35_0 ...
installing: pycrypto-2.6.1-py35_4 ...
installing: pip-8.1.2-py35_0 ...
installing: wheel-0.29.0-py35_0 ...
installing: setuptools-23.0.0-py35_0 ...
Python 3.5.2 :: Continuum Analytics, Inc.
creating default environment...
installation finished.
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /Users/sly/.pyenv/versions/miniconda3-4.1.11:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-env-2.6.0            |                0          601 B
    conda-4.2.13               |           py35_0         389 KB
    pip-9.0.1                  |           py35_1         1.7 MB
    ------------------------------------------------------------
                                           Total:         2.1 MB

The following packages will be UPDATED:

    conda:     4.1.11-py35_0 --> 4.2.13-py35_0
    conda-env: 2.5.2-py35_0  --> 2.6.0-0      
    pip:       8.1.2-py35_0  --> 9.0.1-py35_1 

Fetching packages ...
conda-env-2.6. 100% |####################################################################################################################################| Time: 0:00:00 723.95 kB/s
conda-4.2.13-p 100% |####################################################################################################################################| Time: 0:00:00   1.22 MB/s
pip-9.0.1-py35 100% |####################################################################################################################################| Time: 0:00:00   3.62 MB/s
Extracting packages ...
[      COMPLETE      ]|#######################################################################################################################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|#######################################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|#######################################################################################################################################################| 100%
ERROR: invalid Python executable: /Users/sly/.pyenv/versions/miniconda3-4.1.11/bin/python

The python-build could not find proper executable of Python after successful build.
Please open an issue for future improvements.
https://github.com/yyuu/pyenv/issues

BUILD FAILED (OS X 10.12.2 using python-build 20160602)

Inspect or clean up the working tree at /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229185759.9119
Results logged to /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229185759.9119.log

Last 10 log lines:
installing: yaml-0.1.6-0 ...
installing: zlib-1.2.8-3 ...
installing: conda-4.1.11-py35_0 ...
installing: pycrypto-2.6.1-py35_4 ...
installing: pip-8.1.2-py35_0 ...
installing: wheel-0.29.0-py35_0 ...
installing: setuptools-23.0.0-py35_0 ...
Python 3.5.2 :: Continuum Analytics, Inc.
creating default environment...
installation finished.
```

To get this far, I have to start the installation of either and then go into ~/.pyenv/versions and delete the sym-linked folder while miniconda or anaconda are being downloaded. If I don't delete that folder then this error appears.

```
bash-3.2$ pyenv install -v miniconda3-4.1.11
/var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099 ~
/var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099/Miniconda3-4.1.11-MacOSX-x86_64 /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099 ~
Installing Miniconda3-4.1.11-MacOSX-x86_64...
ERROR: File or directory already exists: /Users/sly/.pyenv/versions/miniconda3-4.1.11

BUILD FAILED (OS X 10.12.2 using python-build 20160602)

Inspect or clean up the working tree at /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099
Results logged to /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099.log

Last 10 log lines:
/var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099 ~
/var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099/Miniconda3-4.1.11-MacOSX-x86_64 /var/folders/vn/6yt22cds5nq4s0vh6gxgzt5r0000gn/T/python-build.20161229190242.10099 ~
ERROR: File or directory already exists: /Users/sly/.pyenv/versions/miniconda3-4.1.11
```

Any help with these problems would be highly appreciated.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/40479671-can-t-install-any-python-version-from-continuum-miniconda-anaconda?utm_campaign=plugin&utm_content=tracker%2F282009&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282009&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/plugins/python-build/bin/python-build
+++ b/plugins/python-build/bin/python-build
@@ -1027,7 +1027,7 @@ anaconda_architecture() {
 
 build_package_anaconda() {
   local package_name="$1"
-  { bash "${package_name}.sh" -b -p "${PREFIX_PATH}"
+  { bash "${package_name}.sh" -f -b -p "${PREFIX_PATH}"
   } >&4 2>&1
 }
 

